pkgs/servers/mobilizon/package.json

```diff
@@ -0,0 +1,100 @@
+{
+  "name": "mobilizon",
```

Comment:
Just a thought, but do we need the package.json?
Is that because they don't commit it in the upstream repo?
I'm just a little concern about committing package.json as it might require freqent changes.

Reply:
It seems necessary for nixpkgs, unfortunately. This was brought to my attention by the ofborg eval failure, before theses changes `mkYarnPackage` would import the package.json from the Mobilizon source at evaluation time, parse the lock file to generate a `.nix` file and import that `.nix` file, also at evaluation time. This would cause some issues if the Mobilizon source derivation is not yet in the Nix store (which seemed to be the case for the ofborg failure), and the `.nix` gymnastic seems to also cause trouble (I assume because you also have to realize a derivation before you can import it).

I checked other Nix packages that use `mkYarnPackage` in nixpkgs, and all of them provide their own `package.json` copied from upstream, and their own pre-generated `yarn.nix` file.

Reply:
I also quite dislike having to copy the `package.json` and having to regenerate the  `yarn.nix` each update, but it seems necessary :confused:

Reply:
I wonder if we need to do some special thing for the update bot to take care of that

Reply:
did you consider just using a mkDerivation and just using a yarn install inside?
It might not be the recommended way, but if the cost of mkYarnPackage are that high, it could be worth considering.
To be sure I'm thinking about something in the vein of
```
  frontendAssets = stdenvNoCC.mkDerivation {
    pname = "frontend-assets-${pname}";
    nativeBuildInputs = [ nodejs nodePackages.pnpm cacert git ];

    inherit version src;

    configurePhase = ''
      cd assets
      pnpm install
    '';

    buildPhase = ''
      pnpm run deploy
      cd ..
    '';

    installPhase = ''
      cp -r priv/static $out
    '';

    outputHashAlgo = "sha256";
    outputHashMode = "recursive";
    outputHash = "sha256-wxVOFq+K8j4lnL7Jx7MXxWoPDf5coJVh8Zb8LFIgjxY=";

    impureEnvVars = lib.fetchers.proxyImpureEnvVars;
  };
```

replace the pnpm parts with yarn of course.
I dont know if with yarn it actually results in a fixed output derivation.

Reply:
I'd like to avoid fixed output derivations if possible. The Yarn2Nix tool is also the common way in nixpkgs to package yarn software, it knows much better than me how to compile / install, and will receive care from other maintainers if something arises.

Reply:
I just realised, that mkYarnPackage doesn't require any of package.json, yarn.lock and yarn.nix to be accessible at eval time (anymore). Even yarn.nix can be generated at build time! I changed this and made the package a lot lighther this way.